### PR TITLE
Added rich data panel link in datamapper

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
* Added rich data panel link in datamapper
* Added GUI test
* Show link only if there is valid theme count or profile data for current geography
* Remove link panel in case of loading state
* clicking on link should open rich data view

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-381

## How is it tested automatically?
Added a GUI test

## How should a reviewer test it locally
Go to Datamapper
If there is valid data in rich data view we show a link and clicking that link should open rich data panel
If there is no data in rich data panel link should be hidden

We can test it on Profile : YE (8) and GDC (11)

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/elements/data_mapper/components/richDataPanelLink.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ]
+  )
+
+  return (
+    <ThemeProvider theme={theme}>
+      <ParentContainer>
+        <IconContainer>
+          <PanelIconSidebar />
+        </IconContainer>
+        <TextContainer>
```
Comment (michaelglenister) [inline src/js/elements/data_mapper/components/richDataPanelLink.js:30]: So this is like Reacts version of CSS class

**Review by michaelglenister — COMMENTED** [review]
I'm happy with the changes so far but I feel concerned about the test data in general. Is there an option to have a reusable data factory with base datasets that can be loaded and modified where needed and is that worth spending time setting up?

File: src/js/elements/data_mapper/components/richDataPanelLink.js (hunk 1 above)
Comment (goyal1092) [reply src/js/elements/data_mapper/components/richDataPanelLink.js:30]: yes it is MUI theme based custom react components.
Regarding reusable data factory: We should divide that into two separate tasks
1) Spike to check available and most compatible factory library
2) Implementation of that library and removing all json files.

I think this one deserves epic of it's own

**Review by michaelglenister — APPROVED** [review]
Approved

**Pull request merged**